Code/Mantid/Framework/Algorithms/inc/MantidAlgorithms/MonitorEfficiencyCorUser.h

```diff
@@ -0,0 +1,56 @@
+#ifndef MANTID_ALGORITHMS_MONITOREFFICIENCYCORUSER_H_
+#define MANTID_ALGORITHMS_MONITOREFFICIENCYCORUSER_H_
+
+#include "MantidKernel/System.h"
+#include "MantidAPI/Algorithm.h"
+
+namespace Mantid
+{
+  namespace Algorithms
+  {
+
+class DLLExport MonitorEfficiencyCorUser : public API::Algorithm
+{
+public:
+  /// (Empty) Constructor
+  MonitorEfficiencyCorUser();// : Mantid::API::Algorithm() {}
+  /// Virtual destructor
+  virtual ~MonitorEfficiencyCorUser();
+  /// Algorithm's name
+  virtual const std::string name() const { return "MonitorEfficiencyCorUser"; }
+  ///Summary of algorithms purpose
+  virtual const std::string summary() const {return "This algorithm normalises the counts by the monitor counts with additional efficiency correction according to the formula set in the instrument parameters file.";}
+
+  /// Algorithm's version
+  virtual int version() const { return 1; }
+  /// Algorithm's category for identification
+  virtual const std::string category() const { return "CorrectionFunctions\\NormalisationCorrections"; }
+
+private:
+  /// Initialisation code
+  void init();
+  ///Execution code
+  void exec();
+
+  double calculateFormulaValue(const std::string&,double);
+  std::string getValFromInstrumentDef(const std::string&);
+  void applyMonEfficiency(const size_t numberOfChannels,
+          const MantidVec& yIn, const MantidVec& eIn, const double coeff,
+          MantidVec& yOut, MantidVec& eOut);
+
+  /// The user selected (input) workspace
+  API::MatrixWorkspace_const_sptr m_inputWS;
```

Comment:
Lots of people use member variables to cache the outputs of properties. It's a practice that I don't particularly like. I think you could do away with these member variables and keep local variables scoped to ::exec, that you can pass into other sub-methods.